docs-guides/source/opt-joint-compression.md

```diff
@@ -0,0 +1,335 @@
+                     minimum_deployment_target=ct.target.macOS15,
+                     )
+mlmodel.save("model_4bit_palettized_with_8bit_quantized_lut.mlpackage")
+```
+
+
+### Joint sparsity and quantization
+
+One way to combining sparsity and quantization is to 
+first quantize a torch model using the Magnitude pruner class, 
+export it as an mlpackage and then apply weight quantization (A16W8) on 
+the mlpackage, as shown in the section above. 
+
+However, if we want to apply pruning and weight-only quantization (A16W8) 
+on the torch model at training time, it can be done as follows.
+ 
+(Note that if `"activation_dtype"` argument in 
+`ModuleLinearQuantizerConfig` is set to `torch.qint8`, 
+its default value, then activations will be quantized to get an A8W8 model) 
+
+```python
+import torchvision 
+import torch
+import coremltools as ct
+from coremltools.optimize.torch.quantization import ModuleLinearQuantizerConfig, \
+                                                    LinearQuantizerConfig, \
+                                                    LinearQuantizer
+
+from coremltools.optimize.torch.pruning import ModuleMagnitudePrunerConfig, \
+                                               MagnitudePrunerConfig, \
+                                               MagnitudePruner 
+                                               
+
+# Initialize model and optimizer
+# e.g. Resnet50
+model = torchvision.models.resnet50(weights="IMAGENET1K_V2")
+model.train()
+optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+
+# Prepare model for joint quantization and pruning
+quant_config = LinearQuantizerConfig(
+    global_config=ModuleLinearQuantizerConfig(
+        quantization_scheme="symmetric",
+        activation_dtype=torch.float,
+    )
+)
+prune_config = MagnitudePrunerConfig(
+    global_config=ModuleMagnitudePrunerConfig(
+         target_sparsity=0.8,       
+    )
+)
+
+
+# The quantizer config needs to be applied before the pruner config
+
+quantizer = LinearQuantizer(model, quant_config)
+quant_model = quantizer.prepare(example_inputs=[1, 3, 256, 256])
+
+pruner = MagnitudePruner(quant_model, prune_config)
+#  in-place is required to ensure quantizer and pruner are 
+# operating on the same model 
+pruned_quant_model = pruner.prepare(inplace=True)
+
+
+n_classes = 1000
+batch_size = 5
+# run a couple of training iterations with random data
+for i in range(2):
+    # Dummy data
+    inputs = torch.randn(batch_size, 3, 256, 256)  # Batch of samples
+    targets = torch.randint(0, n_classes, (batch_size,))  # Target labels 
+    
+    # Forward pass
+    logits = pruned_quant_model(inputs)
+    out = torch.nn.LogSoftmax(dim=1)(logits)
+    loss = torch.nn.functional.nll_loss(out, targets)
+    print(loss)
+    
+    # Backward and optimize
+    optimizer.zero_grad()
+    loss.backward() 
+    optimizer.step()
+    quantizer.step()
+    pruner.step()         
+    
+
+# finalize the model for export
```

Comment:
Should we mention that this ordering is required?

Reply:
Good point, added a comment.